### PR TITLE
Updated examples

### PR DESCRIPTION
**Pull request by lutovich**
PR updates tests that are used in developer manual to highlights recommended API usage and available configuration options.

Depends on https://github.com/neo4j/neo4j-javascript-driver/pull/228 to make `service unavailable example` test pass.

### CODE REVIEW

File: test/v1/examples.test.js [hunk 1]
```diff
@@ -17,421 +17,446 @@
...
-    var s = driver.session();
-    s.run( "CREATE (p:Person {name: {name}})", {name: "The One"} )
-      .then( function(result) {
-        var theOnesCreated = result.summary.counters.nodesCreated();
-        console.log(theOnesCreated);
-        s.close();
-        driver.close();
-      });
+  it('basic auth example', done => {
+    const user = 'neo4j';
```
Comment (zhenlineo) [inline test/v1/examples.test.js]: Can we make `user`, `password` and `uri` global variable instead? As we might change them when migrate from neokit to boltkit.

File: test/v1/examples.test.js [hunk 2]
```diff
@@ -17,421 +17,446 @@
...
+    const principal = 'principal';
+    const credentials = 'credentials';
+    const realm = 'realm';
+    const scheme = 'scheme';
+    const parameters = {};
+
+    // tag::custom-auth[]
+    const driver = neo4j.driver(
+      'bolt://localhost:7687',
+      neo4j.auth.custom(principal, credentials, realm, scheme, parameters)
```
Comment (zhenlineo) [inline test/v1/examples.test.js]: Do me missing a test here?
We could still use `principle=neo4j`, `credentials=neo4j`, `real=local`, `scheme=basic` to use custom auth to pass basic auth to the server.

**Review by zhenlineo — APPROVED** [review]
LG

**Comment by lutovich** [comment]
@zhenlineo rebased and addressed comments

**Comment by zhenlineo** [comment]
LG

**Pull request merged**